packages/zosjobs/src/GetJobs.ts

```diff
@@ -121,55 +134,63 @@ export class GetJobs {
         // TODO(Kelosky): after **REMOVED** is fixed we can remove this message
         Logger.getAppLogger().trace("GetJobs.getJobsCommon()");
         ImperativeExpect.toNotBeNullOrUndefined(session, "Required session must be defined");
         let query = JobsConstants.QUERY_ID;
 
-        if (parms) {
-            if (parms.owner) {
-                query += (JobsConstants.QUERY_OWNER + parms.owner);
+        if (params) {
+            if (params.owner) {
+                query += (JobsConstants.QUERY_OWNER + params.owner);
             }
-            if (parms.prefix) {
-                if (parms.prefix !== JobsConstants.DEFAULT_PREFIX) {
+            if (params.prefix) {
+                if (params.prefix !== JobsConstants.DEFAULT_PREFIX) {
                     if (RestClient.hasQueryString(query)) {
                         query += JobsConstants.COMBO_ID;
                     }
-                    query += JobsConstants.QUERY_PREFIX + parms.prefix;
+                    query += JobsConstants.QUERY_PREFIX + params.prefix;
                 }
             }
-            if (parms.maxJobs) {
-                if (parms.maxJobs !== JobsConstants.DEFAULT_MAX_JOBS) {
+            if (params.maxJobs) {
+                if (params.maxJobs !== JobsConstants.DEFAULT_MAX_JOBS) {
                     if (RestClient.hasQueryString(query)) {
                         query += JobsConstants.COMBO_ID;
                     }
-                    query += (JobsConstants.QUERY_MAX_JOBS + parms.maxJobs);
+                    query += (JobsConstants.QUERY_MAX_JOBS + params.maxJobs);
                 }
             }
-            if (parms.jobid) {
+            if (params.jobid) {
                 if (RestClient.hasQueryString(query)) {
                     query += JobsConstants.COMBO_ID;
                 }
-                query += (JobsConstants.QUERY_JOBID + parms.jobid);
+                query += (JobsConstants.QUERY_JOBID + params.jobid);
             }
-            if (parms.execData) {
+            if (params.execData) {
                 if (RestClient.hasQueryString(query)) {
                     query += JobsConstants.COMBO_ID;
                 }
                 query += (JobsConstants.EXEC_DATA);
             }
+            if (params.status) {
+                if (RestClient.hasQueryString(query)) {
+                    query += JobsConstants.COMBO_ID;
+                }
+                query += (JobsConstants.QUERY_STATUS + params.status);
+            }
         }
 
         let resource = JobsConstants.RESOURCE;
         resource += (query === JobsConstants.QUERY_ID) ? "" : query;
         Logger.getAppLogger().info("GetJobs.getJobsCommon() resource: " + resource);
 
-        return ZosmfRestClient.getExpectJSON<IJob[]>(session, resource);
+        const jobs = await ZosmfRestClient.getExpectJSON<IJob[]>(session, resource);
+
+        return GetJobs.filterResultsByStatuses(jobs, params);
```

Comment:
Do you like one-liners? 😋 

`return params?.status == null || && params.status.toLowerCase() === "active" ? jobs : jobs.filter(job => job.status.toLowerCase() === params.status.toLowerCase())`

I'm not suggesting this though. It'll probably make things a bit confusing 😋

Reply:
Hmmm....one liners are great when checking one condition, can become confusing the more statements we check. The updated filterResultsByStatuses is much simpler now:
```
    private static filterResultsByStatuses(jobs: IJob[], params: IGetJobsParms): IJob[] {
        if (params?.status?.toLowerCase() != "active") {
            return jobs.filter(job => job.status.toLowerCase() === params.status.toLowerCase());
        }
        return jobs;
    }
```